### PR TITLE
feat: Add ability to make generic function calls

### PR DESCRIPTION
**Pull request by Patrick1904**
Reference: https://github.com/near/ui.multisafe/issues/94

This PR adds ability to call generic function calls from the multisafe UI:

Screen Shot 2022-06-20 at 4 17 06 PM

To test:
1. On the left-side panel, click on 'New Transaction'
2. Select 'Make Function Call'
3. Enter the form
4. Submit

### CODE REVIEW

File: src/ui/components/Main/MultiSafe/Sidebar/Actions/NewTransaction/MakeFunctionCall/arguments/Arguments.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { TextField } from '../../../../../../general/TextField/TextField';
+
+export const Arguments = ({ control, classNames, hasError, errorMessage }) => (
+  <TextField
```
Comment (agileurbanite) [inline src/ui/components/Main/MultiSafe/Sidebar/Actions/NewTransaction/MakeFunctionCall/arguments/Arguments.jsx:4]: @Patrick1904 nit: can we make this a text area?

**Review by agileurbanite — CHANGES_REQUESTED** [review]

File: src/store/multisafe/thunks/onMakeFunctionCall.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    {
+      args: {
+        request: {
+          receiver_id: smartContractAddress,
+          actions: [{
+            type: 'FunctionCall',
+            method_name: methodName,
+            args: Buffer.from(args).toString('base64'),
+            deposit: deposit.toString(),
+            gas: tGas.toString(),
```
Comment (judith-Near) [inline src/store/multisafe/thunks/onMakeFunctionCall.js:28]: `const ADD_REQUEST_AND_CONFIRM_GAS = new BN('40000000000000');`
this is 40 tGas

but if a user inputs 300 tGas into the form (the form says it's in tGas), it's only equal to BN('300') in reality

**Review by judith-Near — CHANGES_REQUESTED** [review]
PR lgtm! the only thing that needs to be addressed is the text area for arguments, and the tGas/gas discrepancy

File: src/store/multisafe/thunks/onMakeFunctionCall.js (hunk 2 above)
Comment (Patrick1904) [reply src/store/multisafe/thunks/onMakeFunctionCall.js]: Great catch!

File: src/ui/components/Main/MultiSafe/Sidebar/Actions/NewTransaction/MakeFunctionCall/arguments/Arguments.jsx (hunk 1 above)
Comment (Patrick1904) [reply src/ui/components/Main/MultiSafe/Sidebar/Actions/NewTransaction/MakeFunctionCall/arguments/Arguments.jsx:4]: Good one!

**Review by judith-Near — DISMISSED** [review]
lgtm!

**Review by judith-Near — DISMISSED** [review]

**Review by agileurbanite — DISMISSED** [review]
LGTM

File: src/store/multisafe/thunks/onMakeFunctionCall.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                        gas: formatTGasValue(tGas),
+                    }]
+                },
+            },
+            gas: ADD_REQUEST_AND_CONFIRM_GAS
+        },
+    );
+};
+
+const signTxByLedger = async (
```
Comment (judith-Near) [inline src/store/multisafe/thunks/onMakeFunctionCall.js:40]: nit: can we destructure the args for this object, it's a bit large

**Review by judith-Near — APPROVED** [review]

**Review by judith-Near — APPROVED** [review]

**Pull request merged**